pages/glossary/distinguishable-styles.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Distinguishable Styles
+key: distinguishable-styles
+unambiguous: true
+objective: true
+---
+
+Two elements have _distinguishable styles_ if one of the following is true:
+
+- they have [distinguishable borders](#distinguishable-borders); or
+- they have [distinguishable box-shadows](#distinguishable-box-shadows); or
+- they have different computed values for the `background-image` property; or
+- they have different computed values for at least one [text style property](#text-style-properties).
```

Comment:
It just struck me that this doesn't seem to work correctly for cases like this:

```html
<style>
  a {
    text-decoration: none;
    border-bottom: 1px solid currentColor;
  }

  a:hover, a:focus {
    border-bottom-width: 2px;
  }
</style>

<a href="foo.html">
  <span>Foo</span>
</a>
```

Per the definition, this link doesn't have a distinguishable style even though it's marked with an underline whose width increases on hover and focus. The style is applied to a descendant element though.

This is something we've seen _a lot_ in the wild, by the way.

Reply:
@kasperisager could you elaborate on why do you think this does not work? I'm assuming that the line would have further content to make the rule applicable. Something like:
```html
<a href="foo.html">
  <span>Foo</span>
</a>
Bar
```

The link has a distinguishing style from the non-link text both without focus/hover (it has an underline, while the remaining text hasn't) and with focus/hover (once again, it has an underline while the remaining text hasn't). It seems to pass the rule, but I might be missing something.

Reply:
@carlosapaduarte My bad, I meant to apply the border to the inner `<span>` element 🤦 I've updated the example accordingly.

Reply:
@kasperisager This is one of the situations that fall under the last assumption. Still, I'll try to come up with a solution for these instances where the element being styled is a descendant of the link element.

Reply:
Ah, I see 👍 I do a think a solution is critical as, as mentioned, this is something I see happen quite often. We've made a very similar assumption in our previous, proprietary engine @siteimprove and have seen a steady stream of false-positive reports. At the very least, I hope that this rule will provide industry consensus on a reasonable assumption.

Reply:
Maybe we can strengthen that assumption by adding that case?
"Through manipulation of the positioning and styles of other elements, including non inherited styles of _ancestor_ elements," => "Through manipulation of the positioning and styles of other elements, including non inherited styles of _ancestor_ elements or styles of _descendant_ elements,"


But it might indeed be possible to strengthen "distinguishing style" by considering not the link itself but also any of its ancestor/descendant that contains the exact same (visible?) text nodes.